### ISSUE TITLE
Error after start of container "Error: Cannot find module '@abandonware/bluetooth-hci-socket'"

### ISSUE THREAD

**Issue by Saddamko** [lordthorzonus/ble2mqtt-gateway#453]
Full log is
```
{"log":"node:internal/modules/cjs/loader:1137\n","stream":"stderr","time":"2024-03-27T18:53:56.462891021Z"}
{"log":"  throw err;\n","stream":"stderr","time":"2024-03-27T18:53:56.463274314Z"}
{"log":"  ^\n","stream":"stderr","time":"2024-03-27T18:53:56.463344588Z"}
{"log":"\n","stream":"stderr","time":"2024-03-27T18:53:56.46337945Z"}
{"log":"Error: Cannot find module '@abandonware/bluetooth-hci-socket'\n","stream":"stderr","time":"2024-03-27T18:53:56.463445012Z"}
{"log":"Require stack:\n","stream":"stderr","time":"2024-03-27T18:53:56.463478573Z"}
{"log":"- /home/node/app/node_modules/@abandonware/noble/lib/hci-socket/hci.js\n","stream":"stderr","time":"2024-03-27T18:53:56.463545128Z"}
{"log":"- /home/node/app/node_modules/@abandonware/noble/lib/hci-socket/bindings.js\n","stream":"stderr","time":"2024-03-27T18:53:56.463578031Z"}
{"log":"- /home/node/app/node_modules/@abandonware/noble/lib/resolve-bindings.js\n","stream":"stderr","time":"2024-03-27T18:53:56.463645129Z"}
{"log":"- /home/node/app/node_modules/@abandonware/noble/with-custom-binding.js\n","stream":"stderr","time":"2024-03-27T18:53:56.463676996Z"}
{"log":"- /home/node/app/node_modules/@abandonware/noble/index.js\n","stream":"stderr","time":"2024-03-27T18:53:56.463749269Z"}
{"log":"- /home/node/app/dist/infra/ble-scanner.js\n","stream":"stderr","time":"2024-03-27T18:53:56.463779867Z"}
{"log":"- /home/node/app/dist/gateways/ble-gateway.js\n","stream":"stderr","time":"2024-03-27T18:53:56.463848381Z"}
{"log":"- /home/node/app/dist/index.js\n","stream":"stderr","time":"2024-03-27T18:53:56.463877415Z"}
{"log":"    at Module._resolveFilename (node:internal/modules/cjs/loader:1134:15)\n","stream":"stderr","time":"2024-03-27T18:53:56.46394513Z"}
{"log":"    at Module._load (node:internal/modules/cjs/loader:975:27)\n","stream":"stderr","time":"2024-03-27T18:53:56.464008483Z"}
{"log":"    at Module.require (node:internal/modules/cjs/loader:1225:19)\n","stream":"stderr","time":"2024-03-27T18:53:56.464042367Z"}
{"log":"    at require (node:internal/modules/helpers:177:18)\n","stream":"stderr","time":"2024-03-27T18:53:56.464104999Z"}
{"log":"    at Object.\u003canonymous\u003e (/home/node/app/node_modules/@abandonware/noble/lib/hci-socket/hci.js:6:28)\n","stream":"stderr","time":"2024-03-27T18:53:56.464138692Z"}
{"log":"    at Module._compile (node:internal/modules/cjs/loader:1356:14)\n","stream":"stderr","time":"2024-03-27T18:53:56.464206205Z"}
{"log":"    at Module._extensions..js (node:internal/modules/cjs/loader:1414:10)\n","stream":"stderr","time":"2024-03-27T18:53:56.464238903Z"}
{"log":"    at Module.load (node:internal/modules/cjs/loader:1197:32)\n","stream":"stderr","time":"2024-03-27T18:53:56.464305133Z"}
{"log":"    at Module._load (node:internal/modules/cjs/loader:1013:12)\n","stream":"stderr","time":"2024-03-27T18:53:56.464337542Z"}
{"log":"    at Module.require (node:internal/modules/cjs/loader:1225:19) {\n","stream":"stderr","time":"2024-03-27T18:53:56.464403512Z"}
{"log":"  code: 'MODULE_NOT_FOUND',\n","stream":"stderr","time":"2024-03-27T18:53:56.464434776Z"}
{"log":"  requireStack: [\n","stream":"stderr","time":"2024-03-27T18:53:56.464500044Z"}
{"log":"    '/home/node/app/node_modules/@abandonware/noble/lib/hci-socket/hci.js',\n","stream":"stderr","time":"2024-03-27T18:53:56.46452999Z"}
{"log":"    '/home/node/app/node_modules/@abandonware/noble/lib/hci-socket/bindings.js',\n","stream":"stderr","time":"2024-03-27T18:53:56.464595951Z"}
{"log":"    '/home/node/app/node_modules/@abandonware/noble/lib/resolve-bindings.js',\n","stream":"stderr","time":"2024-03-27T18:53:56.464626344Z"}
{"log":"    '/home/node/app/node_modules/@abandonware/noble/with-custom-binding.js',\n","stream":"stderr","time":"2024-03-27T18:53:56.464693874Z"}
{"log":"    '/home/node/app/node_modules/@abandonware/noble/index.js',\n","stream":"stderr","time":"2024-03-27T18:53:56.464724149Z"}
{"log":"    '/home/node/app/dist/infra/ble-scanner.js',\n","stream":"stderr","time":"2024-03-27T18:53:56.464796691Z"}
{"log":"    '/home/node/app/dist/gateways/ble-gateway.js',\n","stream":"stderr","time":"2024-03-27T18:53:56.46482569Z"}
{"log":"    '/home/node/app/dist/index.js'\n","stream":"stderr","time":"2024-03-27T18:53:56.464968535Z"}
{"log":"  ]\n","stream":"stderr","time":"2024-03-27T18:53:56.465000843Z"}
{"log":"}\n","stream":"stderr","time":"2024-03-27T18:53:56.465177041Z"}
{"log":"\n","stream":"stderr","time":"2024-03-27T18:53:56.465216786Z"}
{"log":"Node.js v18.19.0\n","stream":"stderr","time":"2024-03-27T18:53:56.465285551Z"}

```

**Comment by lordthorzonus** [comment]
Hi thanks for the report! Which version of the container were you trying out? 0.12.0 should be stable at least 👍  I'll have a closer look at this this weekend 💪

**Comment by lordthorzonus** [comment]
There seems to been an issue with the newest noble https://github.com/abandonware/noble/pull/326 so I assume the edge build could contain problems currently as that contains a new version of the noble 🤔 . They seem to have resolved the issue so I'll try to rebuild and release a new version when I can test this out at home 👍. 0.12.0 is using the older noble so that should work currently fine

**Comment by lordthorzonus** [comment]
The issue above proved to be the root cause 👍 Thanks for bringing this up. I just released 0.13.0 https://hub.docker.com/layers/lordthorzonus/ble2mqtt-gateway/0.13.0/images/sha256-d7c0d814611af68618065da1ab4ddd1b97c2ab69dddc9f6be69f00d267969e99?context=explore which fixes the issue. 

The latest edge is also fixed so pulling that tag again should resolve your issue. The edge tag is considered a tad unstable (I now documented this also to the README) so if you don't want to be a beta tester I would recommend using the numbered releases 😁 

Thanks again for reporting this!
